src/main/java/mat/server/service/impl/MatContextServiceUtil.java

```diff
@@ -68,24 +68,26 @@ public boolean isCurrentMeasureEditable(MeasureDAO measureDAO, String measureId)
      */
     public boolean isCurrentMeasureEditable(MeasureDAO measureDAO, String measureId, boolean checkForDraft) {
 
+        String currentUserId = LoggedInUserUtil.getLoggedInUser();
         Measure measure = measureDAO.find(measureId);
+        ShareLevel shareLevel = measureDAO.findShareLevelForUser(currentUserId, measure.getMeasureSet().getId());
+        return isCurrentMeasureEditable(measure, currentUserId, shareLevel == null ? null : shareLevel.getId(), checkForDraft);
+    }
+
+    public boolean isCurrentMeasureEditable(Measure measure, String currentUserId, String shareLevelId, boolean checkForDraft) {
         boolean isEditable = false;
 
-        String currentUserId = LoggedInUserUtil.getLoggedInUser();
         String userRole = LoggedInUserUtil.getLoggedInUserRole();
         boolean isSuperUser = SecurityRole.SUPER_USER_ROLE.equals(userRole);
-        MeasureShareDTO dto = measureDAO.extractDTOFromMeasure(measure);
-        boolean isOwner = currentUserId.equals(dto.getOwnerUserId());
-        ShareLevel shareLevel = measureDAO.findShareLevelForUser(measureId,
-                currentUserId, dto.getMeasureSetId());
+        boolean isOwner = currentUserId.equals(measure.getOwner() == null ? null : measure.getOwner().getId());
         boolean isSharedToEdit = false;
-        if (shareLevel != null) {
-            isSharedToEdit = ShareLevel.MODIFY_ID.equals(shareLevel.getId());
+        if (shareLevelId != null) {

```

Comment:
isSharedToEdit = StringUtils.equals(shareLevelId,ShareLevel.MODIFY_ID) is cleaner imo. You don't need the if

Reply:
Oh.. Good catch. That's a leftover of the old logic when it was an enum.

MODIFY_ID is never null, so a regular "equals" method is sufficient

Reply:
Ah right my bad, thought shareLevel was a string but it is an object.